### DISCUSSION TITLE
知识库检索结果使用分片结果还是Q&A？

### DISCUSSION THREAD

**Discussion by Avanie** [labring/FastGPT#839] [General]
从产品设计来看，若用户采取Q&A入库的方式去进行入库，那么知识库检索就会把大模型提取的Q&A作为引文给模型。实际上大模型提取的Q&A是有幻觉的，建议可以通过Q&A进行检索，然后将Q&A关联的切片原文作为引文。实践中用Q&A作为引文的效果确实没有以原文给大模型的效果好。

**Comment by c121914yu** [comment]
嗯呢，之前 qa 是为了严格场景做的，并不是为了原文检索，只有 5% 的时候需要用。
新的预训练模式会有类似功能。

**Comment by Brender1925** [comment]
个人建议，可以自定义Q&A的提示词，这样的灵活性会更好。
